src/datastore.jl

```diff
@@ -108,7 +108,20 @@ function _enqueue_work(f, args...)
             end
         end
     end
-    put!(SEND_QUEUE, (f, args))
+    if gc_context
+        while true
+            if trylock(SEND_QUEUE)
+                try
+                    put!(SEND_QUEUE, (f, args))
+                finally
+                    unlock(SEND_QUEUE)
+                    break
```

Comment:
Shouldn't this `break` come immediately after the `put!`, so that if we somehow fail to put a new value, we try again? The finally should still be executed even in the presence of `break` (codegen magic).

Reply:
Makes sense, but then maybe we should do the same for the gc_context=false case as well, because that one doesn't have a retry mechanism as well

Reply:
Ok, tested with break moved and it's fine.